### PR TITLE
Bump go to v1.21

### PR DESCRIPTION
**Pull request by robshelly**
### Summary

Update Go to Version 1.21.
 This [commit](https://github.com/package-operator/package-operator/commit/50e01a93171d9036cebfbd38ea72bdff67bb1ebd#diff-0d20884d33c3fe4132b4dddbb12a73fa7934bd38e0ca73865357683dbbc74d8f) introduced the `slices` package which was introduced to Golang in v.1.21.

https://github.com/package-operator/package-operator/blob/58d4aaf9cd4cc8bdfcd19455c21a82c26a3eec8b/internal/packages/internal/packagevalidation/validation.go#L6

### Check List Before Merging

- [ ] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

**Comment by eqrx** [comment]
- Please use the version `1.21.3` since `1.21.2´ contains a CVE
- Please also update the go.mod version of the submodules `apis/` and `pkg/` 
- Please let the github ci use 1.21.3

**Review by eqrx — CHANGES_REQUESTED** [review]
Please use the version 1.21.3 since `1.21.2´ contains a CVE
Please let the github ci use 1.21.3

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module package-operator.run
 
-go 1.20
+go 1.21
```
Comment (eqrx) [inline go.mod:3]:
```suggestion
go 1.21.3
```

File: apis/go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module package-operator.run/apis
 
-go 1.20
+go 1.21
```
Comment (eqrx) [inline apis/go.mod:3]:
```suggestion
go 1.21.3
```

File: go.work [hunk 3]
```diff
@@ -1,4 +1,4 @@
-go 1.20
+go 1.21
```
Comment (eqrx) [inline go.work:1]:
```suggestion
go 1.21.3
```

File: pkg/go.mod [hunk 4]
```diff
@@ -1,6 +1,6 @@
 module package-operator.run/pkg
 
-go 1.20
+go 1.21
```
Comment (eqrx) [inline pkg/go.mod:3]:
```suggestion
go 1.21.3
```

**Review by eqrx — CHANGES_REQUESTED** [review]

File: apis/go.mod (hunk 2 above)
Comment (eqrx) [reply apis/go.mod:3]: (<1.21.3 contains a cve)

**Comment by robshelly** [comment]
Thanks @eqrx. I've added those changes.

**Review by eqrx — APPROVED** [review]

**Pull request merged**